ignite/handlers/__init__.py

```diff
@@ -43,6 +52,13 @@
     "create_lr_scheduler_with_warmup",
     "FastaiLRFinder",
     "EMAHandler",
+    "BaseParamScheduler",
+    "StatePScheduler",
+    "LambdaStatePScheduler",
+    "LinearStatePScheduler",
+    "ExpStatePScheduler",
+    "StepStatePScheduler",
+    "MultiStepStatePScheduler",
```

Comment:
Are we happy with these names ? @sdesrozis 
I'm hesitating between: StatePScheduler and SParamScheduler ...

Reply:
For optimizer parameter's scheduler, the base class is `ParamScheduler` but the inherited classes are like `LinearScheduler` or `CyclicalScheduler` => `Param` has been removed.

If the base class is now `StateParamScheduler`, following the same rule, the inherited could be `CyclicalStateScheduler`, yet it doesn't sound very well...

I think, we could have something like `OptimParamScheduler` and `StateParamScheduler` as base classes, and it could lead to `CyclicalOPScheduler` and `CyclicalSPScheduler`. We need something similar for both cases IMO.

Anyway, I don't know how choose between `StatePScheduler` and `SParamScheduler`...

Reply:
I agree It would be better to rename base optimizer classes, but don't you think it would break  BC ?

Reply:
Renaming such a class will surely break BC.

Firstly, let's choose between `xxxStatePScheduler` or `xxxSParamScheduler` or better. For optimizer, the rule is `xxxScheduler` => Why not `xxxStateScheduler` ?

Reply:
> Why not `xxxStateScheduler` ?

Sounds good to me

Reply:
ok for `xxxStateScheduler `